js/script.js

```diff
@@ -1,88 +1,100 @@
-
-    rgbColor = baseColor();
-    html = `<div style="background-color: ${rgbColor}" class="darkmode-ignore colors"></div>`
-
-    print( `<section class="card"></br> <p>rgb(${red}, ${green}, ${blue})</p> </br> ${html} </br></section>` );
-
-    red += 2;
-    blue += 2;
-    green += 2;
-  }
-
-  print('</section>');
-
+const getRGBValues = colors => {
+    return colors.map(color => {
+        return parseInt(document.getElementsByClassName(color)[0].value, 10);
+    });
+};
+
+const isValid = value => !isNaN(value) && value < 256 && value >= 0;
+
+const valuesValidators = (colors, values) => {
+    colors.forEach((color, index) => {
+        const element = document.getElementsByClassName(color)[0].classList;
+        element.remove("error");
+        !isValid(values[index]) && element.add("error");
+    });
+};
+
+const baseColor = ({ red, blue, green }) => {
+    return `rgb(${red}, ${green}, ${blue})`;
+};
+
+const showAlert = () => {
+    const hasErrors = document.getElementsByClassName("error").length;
+    return (
+        hasErrors &&
+        alert("Make sure all color inputs are valid numbers between 0 and 255.")
+    );
+};
+
+const resetShades = () => {
+    const addJs = document.getElementById("add-js");
+    while (addJs.hasChildNodes()) {
+        addJs.removeChild(addJs.firstChild);
+    }
+};
+
+const printContent = content => {
+    document.getElementById("add-js").innerHTML += `${content}`;
+};
+
+const addTitle = (type = "base") => {
+    const title = {
+        base: `<h2>Here are your results:</h2><h3>Base Color:</h3>`,
+        shade: `<h3>Shades:</h3>`
+    };
+
+    return title[type];
+};
+
+const addCard = ({ red, green, blue }) => {
+    const rgbColor = baseColor({ red, green, blue });
+
+    return `
+        <section class="card">
+            <p>rgb(${red}, ${green}, ${blue})</p>
+            <div>
+                <div 
+                    style="background-color: ${rgbColor}" 
+                    class="darkmode-ignore colors"
+                />
+            </div>
+        </section>
+    `;
+};
+
+document.getElementById("submit").addEventListener("click", event => {
+    //removes any previously generated shade Sets
+    resetShades();
+
+    let content = "";
+    const colors = ["red", "blue", "green"];
+    const values = getRGBValues(colors);
+    let [red, green, blue] = values;
+
+    valuesValidators(colors, values);
+    showAlert();
+
+    //Initial Color Print
+    content += addTitle("base");
+    content += addCard({ red, green, blue });
+    content += addTitle("shade");
+
+    //Sets color to the darkest Shade
+    while (red > 0 && green > 0 && blue > 0) {
+        red--;
+        green--;
+        blue--;
+    }
+
+    //Prints all Shades
+
+    while (red < 256 && green < 256 && blue < 256) {
+        content += addCard({ red, green, blue });
+
+        red += 2;
+        blue += 2;
+        green += 2;
+    }
+
+    printContent(content);
```

Comment:
These revisions definitely help in terms of code clarity and separation of data from the code. I think the organization and optimization of the code works really nicely as well. One thing that would be nice is clearer comments in the first section of code, however. Once I studied the code for a bit, it became pretty clear, but comment would certainly help to initially orient other developers to what's going on. Other than that, this look really nice! As always, thanks for your contribution.

Reply:
Sure. I could add a blurb or two, but I feel like the function names are very descriptive as to their purpose.

RE: the huge margins on the cards, what do you thing of having the cards just be the width of animated colored box and without the `rgb()` text -- except maybe the base colour? In place of the text a tooltip would display it on hover. That way more cards can be shown be shown per row, they can be arranged using flexbox/grid for different size screens, and reduce the amount of vertical scrolling -- especially when there are a __lot__ of shades.

Reply:
Hmm... I like the idea of removing the rgb code from above the color, but I’m thinking it might be better to have the user be able to click on the card and have the card expand to cover the screen and show the rgb code. The reason I say this is because while tooltips are great, they only work will really small chunks of text. At this point, the rgb code isn’t that big, but I have plans to add things like hex values and the like in the future, which would require more space. Creating something that is easy to add on to at this point seems like the best option for me, but I’d love to hear your thoughts on any other ideas. Also, I’m trying to keep the design look minimalistic, so while more shades per row is nice, I think some extra margin on either side of the page will be beneficial from a design viewpoint. Using flexbox or grid certainly sounds like a great way to do this. However, at this point I believe as separate issue and PR is needed, so I’ll open that up with some of the details we discussed if you approve.